packages/nx/src/lock-file/yarn.ts

```diff
@@ -288,6 +297,51 @@ function pruneDependencies(
   return result;
 }
 
+function pruneDependency(
+  packageName: string,
+  normalizedPackageJson: PackageJsonDeps,
+  dependencies: LockFileData['dependencies'],
+  result: LockFileData['dependencies'],
+  isPatch = false
+) {
+  const [key, value] = Object.entries(dependencies[packageName]).find(
+    ([, v]) => v.rootVersion
+  );
+
+  result[packageName] = result[packageName] || {};
+  if (isPatch) {
+    const originalPackageName = packageName.split('@patch:').pop();
+    const patchSuffix = value.packageMeta[0].split('#').pop();
+    value.packageMeta = [
+      `${packageName}@${getVersionFromPackageJson(
```

Comment:
For patch we take patch name + version of the original package in package.json + patch suffix